### PR TITLE
Fix numpy deprecation warning when using array with single entry as a scalar

### PR DESCRIPTION
**Pull request by Luthaf**
This also removes some `try/catch` blocks, which will help with TorchScript support (#280 is another step in this direction).

----
:books: Documentation preview :books:: https://equistore--292.org.readthedocs.build/en/292/

### CODE REVIEW

File: python/equistore-operations/tests/multiply.py [hunk 1]
```diff
@@ -266,14 +266,11 @@ def test_self_multiply_scalar_gradient(self):
         keys = Labels(names=["key_1", "key_2"], values=np.array([[0, 0], [1, 0]]))
         A = TensorMap(keys, [block_1, block_2])
         B = 5.1
-        C = np.array([5.1])
```
Comment (DavideTisi) [inline python/equistore-operations/tests/multiply.py:269]: is this "feature" removed now?

File: python/equistore-operations/tests/pow.py [hunk 2]
```diff
@@ -111,14 +111,10 @@ def test_self_pow_scalar_gradient(self):
 
         keys = Labels(names=["key_1", "key_2"], values=np.array([[0, 0], [1, 0]]))
 
-        C = np.array([2.0])
```
Comment (DavideTisi) [inline python/equistore-operations/tests/pow.py:114]: same as multiply

File: python/equistore-operations/tests/subtract.py [hunk 3]
```diff
@@ -264,14 +264,11 @@ def test_self_subtract_scalar_gradient(self):
         keys = Labels(names=["key_1", "key_2"], values=np.array([[0, 0], [1, 0]]))
         A = TensorMap(keys, [block_1, block_2])
         B = -5.1
-        C = np.array([-5.1])
```
Comment (DavideTisi) [inline python/equistore-operations/tests/subtract.py:267]: same as before

**Review by DavideTisi — COMMENTED** [review]
Everything is fine, did we remove the feature that we can use `np.array([[scalar]])` ?

**Comment by Luthaf** [comment]
> Everything is fine, did we remove the feature that we can use np.array([[scalar]]) ?


Yes, because it is deprecated in numpy 1.25:

```
>>> import numpy as np
>>> float(np.array([3.2]))
<stdin>:1: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
3.2
```

**Pull request merged**